cmd/dep/status.go

```diff
@@ -40,9 +41,43 @@ print an extended status output for each dependency of the project.
   TODO    Another column description
   FOOBAR  Another column description
 
+You may use the -f flag to create a custom format for the output of the
+dep status command. The available fields you can utilize are as follows:
+ProjectRoot, Constraint, Version, Revision, Latest, and PackageCount.
+
 Status returns exit code zero if all dependencies are in a "good state".
 `
 
+const statusExamples = `
+dep status
+
+	Displays a table of the various dependencies in the project along with
+	their properties such as the constraints they are bound by and the 
+	revision they are at.
+
+dep status -f='{{if eq .Constraint "master"}}{{.ProjectRoot}} {{end}}'
+
+	Display the list of package names constrained on the master branch.
+	The -f flag allows you to use Go templates along with it's various 
+	constructs for formating the output data. See -help for available 
+	variables for this flag.
```

Comment:
Small UX concern: I'd re-list the available variables here instead of making the user run a separate command.  For most people trying to use this feature, I expect they'd want to see the example and the options side-by-side.  

To keep it nice and DRY, a simple `const templateVariablesLine = "ProjectRoot, Constraint, Version, Revision, Latest, and PackageCount.\n"` could be used in both places.

Reply:
I guess if we can put in a line of description telling the user where to go to see the variable names, we can instead add a line telling what those actually are. 😅